Xamarin.Forms.Platform.iOS/Renderers/ShellRenderer.cs

```diff
@@ -195,6 +195,7 @@ protected override void Dispose(bool disposing)
 			{
 				_disposed = true;
 				FlyoutRenderer?.Dispose();
+				SetCurrentShellItemController(null);
```

Comment:
When the main page was getting swapped out the previous shell instance is sticking around because it's not getting cleaned up